### PR TITLE
Cypress testing suite

### PR DESCRIPTION
**Pull request by jamopopper**
The cypress testing suite works on my branch, time to see if it works on main!!

### CODE REVIEW

File: cypress/e2e/spec.cy.js [hunk 1]
```diff
@@ -1,6 +1,58 @@
...
+
+it('Data is saved between tabs', function () {
+    cy.visit('/')
+    cy.findAllByText("Line").click()
+    cy.findAllByLabelText("Chart title").type('Cat super powers')
+    cy.findAllByLabelText("X label").type('cats')
+    cy.findAllByLabelText("Y label").type('powers')
+    cy.findAllByLabelText("X").type('6')
+    cy.findAllByLabelText("Y").type('9')
+    cy.findAllByLabelText("X").first().invoke('val').should('equal', '6')
```
Comment (elysemc1) [inline cypress/e2e/spec.cy.js:23]: I think you need to click "Scatter" after clicking "Line" and then assert that the data is the same here. Should just be one more line, right after you type everything. Just to make sure that it stays the same when you switch!

File: cypress/e2e/spec.cy.js [hunk 2]
```diff
@@ -1,6 +1,58 @@
...
+    cy.findAllByText("Line").click()
+    cy.findAllByLabelText("Chart title").type('Cat super powers')
+    cy.findAllByLabelText("X label").type('cats')
+    cy.findAllByLabelText("Y label").type('powers')
+    cy.findAllByLabelText("X").type('4')
+    cy.findAllByLabelText("Y").type('2')
+    cy.findAllByText("Generate chart").click()
+    cy.findAllByText("Save chart").click()
+    cy.findAllByText("Gallery").click()
+    cy.get('#gallery').children().should('have.length', 1)
```
Comment (elysemc1) [inline cypress/e2e/spec.cy.js:40]: Could you also assert that the title matches 'Cat super powers'?

File: cypress/e2e/spec.cy.js [hunk 3]
```diff
@@ -1,6 +1,58 @@
...
+    cy.findAllByLabelText("Y label").type('powers')
+    cy.findAllByLabelText("X").type('12')
+    cy.findAllByLabelText("Y").type('21')
+    cy.findAllByText("Generate chart").click()
+    cy.findAllByText("Save chart").click()
+    cy.findAllByText("Gallery").click()
+    cy.get('#gallery').children().click()
+    cy.findAllByLabelText("X").first().invoke('val').should('equal', '12')
+    cy.findAllByLabelText("Y").first().invoke('val').should('equal', '21')
+    cy.findAllByLabelText("X label").parent().parent().children().should('have.length', 7)
```
Comment (elysemc1) [inline cypress/e2e/spec.cy.js:59]: What does this line check?

**Review by elysemc1 — CHANGES_REQUESTED** [review]
Just a few small things!! Overall great through.

File: cypress/e2e/spec.cy.js (hunk 3 above)
Comment (elysemc1) [reply cypress/e2e/spec.cy.js:59]: Line 55, I mean

Comment (jamopopper) [reply cypress/e2e/spec.cy.js:59]: It checks that the correct number of data points are in the graph (should be one set, which makes it equal 7 children things)

Comment (jamopopper) [reply cypress/e2e/spec.cy.js:59]: 2 for the x and y, two for the x and y axis names, 2 for the new blank x and y added by the page when refreshed, and one for the table header

File: cypress/e2e/spec.cy.js (hunk 1 above)
Comment (jamopopper) [reply cypress/e2e/spec.cy.js:23]: you right, I also forgot to generate the plot bc I think it saves it?? idk, but it doesn't hurt it. changed on incoming commit

File: cypress/e2e/spec.cy.js (hunk 2 above)
Comment (jamopopper) [reply cypress/e2e/spec.cy.js:40]: yessir, happens in incoming commit

**Review by elysemc1 — APPROVED** [review]
Thanks for the changes, looks great!

**Pull request merged**